src/modules/navigator/mission.cpp

```diff
@@ -2125,3 +2050,24 @@ void Mission::updateCachedItemsUpToIndex(const int end_index)
 		}
 	}
 }
+
+void Mission::checkClimbRequired()
+{
+	mission_item_s next_position_mission_item = {};
+
+	if (getNextPositionMissionItem(_mission, _current_mission_index, next_position_mission_item)) {
+		const float altitude_amsl_next_position_item = get_absolute_altitude_for_item(next_position_mission_item);
+
+		const float error_below_setpoint = altitude_amsl_next_position_item -
+						   _navigator->get_global_position()->alt;
+
+		if (error_below_setpoint > _navigator->get_altitude_acceptance_radius()) {
+
+			_mission_init_climb_altitude_amsl = altitude_amsl_next_position_item;
+
+		} else {
+
+			_mission_init_climb_altitude_amsl = NAN;
+		}
+	}
```

Comment:
RTL is it's own thing, and doesn't require special handling here. If it's a RTL item it will switch to the RTL mode, which itself has a climb altitude.

In general, if there is no item containing a position after the start/resume index there isn't much to do, the vehicle then should simply stay at the current position.